python_modules/dagster/dagster/_core/definitions/decorators/repository_decorator.py

```diff
@@ -88,8 +99,17 @@ def __call__(self, fn: Callable[[], Any]) -> RepositoryDefinition:
                     "AssetsDefinition, or SourceAsset."
                     f"Got {bad_definitions_str}."
                 )
+            if cacheable_defns:
+                return PendingRepositoryDefinition(
+                    self.name,
+                    repository_definitions=repository_defns,
+                    cacheable_definitions=cacheable_defns,
+                    description=self.description,
+                    default_executor_def=self.default_executor_def,
+                    default_logger_defs=self.default_logger_defs,
+                )
```

Comment:
bit worried about how this branching path will age - this code is already really gnarly having divergence like this makes it worse

Reply:
yeah that's legit -- refactored it a bit such that it doesn't need to branch at that point (although the logic is still fundamentally branching, so the amount of gnarl removed is pretty minimal)